test/lib/ufe/testDisplayLayer.py

```diff
@@ -304,8 +304,51 @@ def testDisplayLayerItemDelete(self):
             self.assertFalse(defaultLayer.contains(self.SPHERE1))
         self.assertFalse(layer1.contains(self.SPHERE1))
 
-        # TODO - test the undo/redo for delete of Ufe item in Layer.
-        # This currently doesn't work.
+    @unittest.skipUnless(mayaUtils.ufeSupportFixLevel() >= 6, "Requires Display Layer Ufe item delete undo fix.")
+    def testDisplayLayerItemDeleteUndo(self):
+        '''
+        Test that undoing a prim deletion restores its display layer inclusion.
+        '''
+        # First create Display Layer and add some prims to it.
+        cmds.createDisplayLayer(name=self.LAYER1, number=1, empty=True)
+        defaultLayer = self.displayLayer(self.DEFAULT_LAYER)
+        layer1 = self.displayLayer(self.LAYER1)
+        psPathStr = mayaUsd_createStageWithNewLayer.createStageWithNewLayer()
+        stage = mayaUsd.lib.GetPrim(psPathStr).GetStage()
+        stage.DefinePrim('/Cube1', 'Cube')
+        stage.DefinePrim('/Sphere1', 'Sphere')
+
+        # Add the two prims to the layer.
+        cmds.editDisplayLayerMembers(self.LAYER1, self.SPHERE1, self.CUBE1, noRecurse=True)
+
+        def verifyInLayer(present, absent):
+            # Verify they are in layer.
+            layerObjs = cmds.editDisplayLayerMembers(self.LAYER1, query=True, **self.kwArgsEditDisplayLayerMembers)
+            for item in present:
+                self.assertTrue(item in layerObjs, "Item %s should be in layer" % item)
+            for item in absent:
+                self.assertFalse(item in layerObjs, "Item %s should not be in layer" % item)
+
+        verifyInLayer({ self.CUBE1, self.SPHERE1 }, { self.NEW_SPHERE1 })
+
+        self._testLayerFromPath(self.CUBE1, self.LAYER1)
+        self._testLayerFromPath(self.SPHERE1, self.LAYER1)
+
+        # Delete the Sphere and make sure it is removed from the layer.
+        cmds.delete(self.SPHERE1)
+
+        verifyInLayer({ self.CUBE1 }, { self.NEW_SPHERE1, self.SPHERE1 })
+
+        # Must issue two undo because delete spawed another command
+        # That updated the display layer. The alternative would be to
+        # open an explicit undo chunk to combine both commands into one undo.
+        cmds.undo()
+        cmds.undo()
```

Comment:
This feels very awkward.  I know you've mentioned that in the interactive application all commands are chunked, so to test what the interactive application is doing I would follow your suggestion and open an explicit undo chunk before line 338.